portiaui/app/models/start-url.js

```diff
@@ -33,41 +67,48 @@ export default function startUrl(spec) {
         });
         return urlList;
     }
+});
 
-    function save(spider) {
-        const urls = spider.get('startUrls');
-        urls.pushObject(urlObject);
-        spider.save();
-        return urlObject;
-    }
 
-    function toString() {
-        if (urlObject.isGenerated) {
-            return urlObject.fragments.map(fragmentToString).join('');
-        } else {
-            return urlObject.url;
+const FeedUrl = StartUrl.extend({
+    type: 'feed',
+    componentName: 'project-structure-spider-feed-url',
+    optionsComponentName: 'feed-url-options',
+
+    show() {
+        return this._raw_url();
+    },
+
+    _raw_url() {
+        const url = this.get('url');
+        const notRaw = !url.includes('raw');
+
+        if (url.includes('gist.github') && notRaw) {
+            const trailingSlash = (url.slice(-1) === '/') ? '' : '/';
+            return url + trailingSlash + 'raw';
         }
-    }
 
-    function serialize() {
-        let base = {
-            'url': urlObject.toString(),
-            'type': urlObject.type
-        };
-        if (urlObject.isGenerated) {
-            base.fragments = urlObject.fragments;
+        if (url.includes('dropbox.com') && notRaw) {
+            return url + '&raw=1';
         }
-        return base;
-    }
 
-    let urlObject = buildUrlObject();
-    urlObject.isGenerated = urlObject.type === 'generated';
+        if (url.includes('google.com') && !url.includes('export')) {
```

Comment:
Bitbucket snippets are tricky to get right. I opted for going with the url `http://files.scrapinghub.com/portia/urls.txt`.